### PR TITLE
[Example] Add with-web-worker example

### LINKED ISSUE #12545: WebWorker example (TypeScript)
# Feature request

## Is your feature request related to a problem? Please describe.

There is currently no example for WebWorker yet.

## Describe the solution you'd like

WebWorker example

## Describe alternatives you've considered

Tried this one : https://github.com/kuwabataK/next-template/tree/service-worker-test
with my project but 

```
Reference Error: Worker is undefined
```

### PR DESCRIPTION
**Pull request by Prioe**
Resolves: https://github.com/zeit/next.js/issues/12545

This PR adds an example which show how to bundle and use WebWorkers in a next.js app.

The issue mentions Typescript, but adding typescript to the example should be fairly simple by following the documentation and seems out of scope for a web-worker example.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12.1s | -201ms |
| nodeModulesSize | 67.3 MB | 67.3 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.089 | 2.066 | -0.02 |
| / avg req/sec | 1196.51 | 1210.19 | ⚠️ +13.68 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.158 | 1.152 | -0.01 |
| /error-in-render avg req/sec | 2158.1 | 2169.23 | ⚠️ +11.13 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..c03c.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.92 kB | 6.92 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| index.html gzip | 927 B | 927 B | ✓ |
| link.html gzip | 936 B | 936 B | ✓ |
| withRouter.html gzip | 925 B | 925 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13.1s | -273ms |
| nodeModulesSize | 67.3 MB | 67.3 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..c03c.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.92 kB | 6.92 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | Prioe/next.js example-with-web-worker | Change |
| - | - | - | - |
| _error.js | 876 kB | 876 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 877 kB | 877 kB | ✓ |
| link.js | 914 kB | 914 kB | ✓ |
| routerDirect.js | 907 kB | 907 kB | ✓ |
| withRouter.js | 907 kB | 907 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>

File: examples/with-web-worker/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# WebWorker example
+
+This example shows how to run code in WebWorkers by utilizing [worker-plugin](https://github.com/GoogleChromeLabs/worker-plugin).
+
+## Deploy your own
+
+Deploy the example using [Vercel](https://vercel.com/now):
```
Comment (Timer) [inline examples/with-web-worker/README.md:7]:
```suggestion
Deploy the example using [Vercel](https://vercel.com/):
```

File: examples/with-web-worker/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/import/project?template=https://github.com/zeit/next.js/tree/canary/examples/with-web-worker)
```
Comment (Timer) [inline examples/with-web-worker/README.md:9]:
```suggestion
[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/import/project?template=https://github.com/vercel/next.js/tree/canary/examples/with-web-worker)
```

File: examples/with-web-worker/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+## How to use
+
+### Using `create-next-app`
+
+Execute [`create-next-app`](https://github.com/zeit/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:
```
Comment (Timer) [inline examples/with-web-worker/README.md:15]:
```suggestion
Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:
```

File: examples/with-web-worker/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# or
+yarn create next-app --example with-web-worker with-web-worker-app
+```
+
+### Download manually
+
+Download the example:
+
+```bash
+curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/with-web-worker
```
Comment (Timer) [inline examples/with-web-worker/README.md:28]:
```suggestion
curl https://codeload.github.com/vercel/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/with-web-worker
```

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]
Thanks!

**Comment by iammarkps** [comment]
Thank you!

**Pull request merged**